specs/BrowserProcessExited.md

```diff
@@ -66,19 +78,33 @@ void AppWindow::CloseWebView(/* ... */) {
               ICoreWebView2Environment* sender,
               ICoreWebView2BrowserProcessExitedEventArgs* args) {
           COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND kind;
+          UINT32 pid;
           CHECK_FAILURE(args->get_BrowserProcessExitKind(&kind));
-
-          // Watch for graceful browser process exit. Let ProcessFailed event
-          // handler take care of failed browser process termination.
-          if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL)
+          CHECK_FAILURE(args->get_BrowserProcessId(&pid));
+
+          // If a new WebView is created from this CoreWebView2Environment after
+          // the browser has exited but before our handler gets to run, a new
+          // browser process will be created and lock the user data folder
+          // again. Do not attempt to cleanup the user data folder in these
+          // cases. We check the PID of the exited browser process against the
+          // PID of the browser process to which our last CoreWebView2 attached.
+          if (pid == m_newestBrowserPid)
```

Comment:
A different instance of the app could be started up that uses the user data folder and that could happen in between this event firing and this line or the next line. Does having the PID of the browser process that closed help if a new browser process could be created at any time? Or is adding this PID part of fixing a different problem?

Reply:
This API is only aware of the context provided by the current environment instance. The PID was added for the case where a new browser process connection could be established before the `BrowserProcessExited` handler has got to run, when the same environment instance creates a new webview after closing all previous ones.